torchrl/collectors/collectors.py

```diff
@@ -848,10 +843,9 @@ def _step_and_maybe_reset(self) -> None:
             if td_reset.batch_dims:
                 # better cloning here than when passing the td for stacking
                 # cloning is necessary to avoid modifying entries in-place
-                self._tensordict = self._tensordict.clone()
-                self._tensordict[traj_done_or_terminated] = td_reset[
-                    traj_done_or_terminated
-                ]
+                self._tensordict = torch.where(
```

Comment:
is this compatible with lazy stacks?

Reply:
[yep](https://github.com/pytorch-labs/tensordict/blob/87299f7069acc3c9a1cc2c53656c6883e7ed8f52/tensordict/tensordict.py#L7586)